src/components/Header/NetworkSelector.tsx

```diff
@@ -308,7 +307,7 @@ export default function NetworkSelector() {
   }
 
   return (
-    <SelectorWrapper ref={node as any} onMouseEnter={toggle} onMouseLeave={toggle}>
+    <SelectorWrapper ref={node} onMouseEnter={openModal} onMouseLeave={closeModal} onClick={toggle}>
```

Comment:
The reason why we can't have just do:

```jsx
 <SelectorWrapper ref={node} onMouseEnter={toggle} onMouseLeave={toggle} onClick={toggle}>
```

is that we could end up in this state:

1. hover to open
2. click to close (at this point, mouse is still hovering over element)

this would mean that when the mouse leaves, the `toggle` function would be called again, thereby reopening the modal (which is not the desired behaviour)